### PR TITLE
Fix svelte adapter issues

### PR DESCRIPTION
**Pull request by Mokshit06**
This PR fixes the following issues in the svelte adapter:
- make the `Placeholder` component work on both server and client
- custom components (cell/footer) work with hydration
- detect svelte components in hmr env and handle them

### CODE REVIEW

**Comment by Mokshit06** [comment]
I think I created this pr from alpha branch, should I close this one and create another one from beta?

**Comment by KevinVandy** [comment]
> I think I created this pr from alpha branch, should I close this one and create another one from beta?

Pretty sure its fine since alpha -> beta when its ready for new beta release, but will hold off merging till @tannerlinsley confirms.

**Review by tannerlinsley — CHANGES_REQUESTED** [review]

File: packages/svelte-table/src/placeholder.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/* App.svelte generated by Svelte v3.48.0 */
+import { create_ssr_component, escape } from 'svelte/internal'
+import PlaceholderClient from './placeholder.svelte'
+
+const PlaceholderServer = create_ssr_component(
+  ($$result, $$props, $$bindings, slots) => {
```
[the remark this answers is not in the document]
Comment (Mokshit06) [reply packages/svelte-table/src/placeholder.ts:6]: I think we can change to what the svelte language server shows. I'll add that commit

**Comment by tannerlinsley** [comment]
Should be fine.

**Pull request merged**